variables.tf

```diff
@@ -45,8 +45,49 @@ variable "plan" {
   }
 }
 
+variable "rotation_enabled" {
+  type        = bool
+  description = "If set to true, Key Protect enables a rotation policy on the instance."
+  default     = true
+}
+
+variable "rotation_interval_month" {
+  type        = number
+  description = "Specifies the key rotation time interval in months. Must be between 1 and 12 inclusive."
+  default     = 1
+
+  validation {
+    condition     = (var.rotation_interval_month >= 1) && (var.rotation_interval_month <= 12)
+    error_message = "The rotation_interval_month must be between 1 and 12 inclusive."
+  }
+}
+
+variable "dual_auth_delete_enabled" {
+  type        = bool
+  description = "If set to true, Key Protect enables a dual authorization policy on the instance. Note: Once the dual authorization policy is set on the instance, it cannot be reverted. An instance with dual authorization policy enabled cannot be destroyed using Terraform."
+  default     = false
+}
+
 variable "metrics_enabled" {
   type        = bool
-  description = "Set to true to enable metrics on the Key Protect instance. In order to view metrics, you will need a Monitoring (Sysdig) instance that is located in the same region as the Key Protect instance. Once you provision the Monitoring instance, you will need to enable platform metrics."
+  description = "If set to true, Key Protect enables metrics on the Key Protect instance. In order to view metrics, you will need a Monitoring (Sysdig) instance that is located in the same region as the Key Protect instance. Once you provision the Monitoring instance, you will need to enable platform metrics."
   default     = true
 }
+
+variable "key_create_import_access_enabled" {
+  type        = bool
+  description = "If set to true, Key Protect enables a key_create_import_access policy on the instance."
+  default     = true
+}
+
+variable "key_create_import_access_settings" {
+  type = object({
+    create_root_key     = optional(bool, true)
+    create_standard_key = optional(bool, true)
+    import_root_key     = optional(bool, true)
+    import_standard_key = optional(bool, true)
+    enforce_token       = optional(bool, false)
+  })
+  description = "Key create import access policy settings to configure if var.enable_key_create_import_access_policy is true. For more info see https://cloud.ibm.com/docs/key-protect?topic=key-protect-manage-keyCreateImportAccess"
+  default     = {}
```

Comment:
I think we want to set a default value here right? Whatever the best practise is

Reply:
The `optional(bool, <true|false>)` in the definition is setting the defaults, we can also pass the default as a map, but the advantage to using the `optional` style is that each of the attributes within the map becomes optional so the entire map doesn't have to be overridden as it does in the case where we pass `default = { create_root_key = true...}`. If it is preferable to require the entire map be passed when overriding defaults it can be changed from the optional method.

Reply:
Oh OK yes I see that in the plan:
```
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:       + key_create_import_access {
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + create_root_key     = true
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + create_standard_key = true
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + created_by          = (known after apply)
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + creation_date       = (known after apply)
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + enabled             = true
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + enforce_token       = false
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + import_root_key     = true
TestRunDefaultExample [202](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect/actions/runs/4132819570/jobs/7141975122#step:7:203)3-02-09T09:31:51Z logger.go:66:           + import_standard_key = true
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + last_updated        = (known after apply)
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:           + updated_by          = (known after apply)
TestRunDefaultExample 2023-02-09T09:31:51Z logger.go:66:         }
```

Reply:
What happens if you remove the default value?

Reply:
In testing the defaults we set I found a bug with the provider not actually using the values you pass and falling back to its own defaults for the policies, created https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4340 for the provider team to address this issue.

Reply:
resolving conversation as mentioned in scrum we will remove the additional instance policies until the provider bug is resolved.